bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxLocalAccessHandler.java

```diff
@@ -162,10 +288,28 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     @Override
     public void dispose() {
         super.dispose();
+        cancelSchedule();
+    }
+
+    private void cancelSchedule() {
         ScheduledFuture<?> schedule = this.schedule;
         if (schedule != null) {
             schedule.cancel(true);
             this.schedule = null;
         }
     }
+
+    private <T extends Quantity<T>> void updateChannel(String channelID, double value, Unit<T> unit,
+            Set<String> supportedChannels) {
+        if (supportedChannels.contains(channelID)) {
+            if (value > Short.MIN_VALUE) {
+                updateState(channelID, new QuantityType<>(value, unit));
+            } else {
+                updateState(channelID, UnDefType.UNDEF);
+                logger.warn(
+                        "Channel {} is marked as supported, but its value is out of the defined range. Value = {}. This is unexpected behaviour. Please file a bug.",
+                        channelID, value);
```

Comment:
You could consider adding the channel to a `Set` which could be checked in order to log this warning only once, not after each call.